/home/zcw7438/.local/lib/python3.6/site-packages/pandas/compat/_optional.py:106: UserWarning: Pandas requires version '1.2.1' or newer of 'bottleneck' (version '1.2.0' currently installed).
  warnings.warn(msg, UserWarning)
sys:1: DtypeWarning: Columns (18,20,28,30,32,34,36,38,55) have mixed types. Specify dtype option on import or set low_memory=False.
/software/anaconda3.6/lib/python3.6/site-packages/numpy/lib/arraysetops.py:569: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)
/home/zcw7438/.local/lib/python3.6/site-packages/pandas/compat/_optional.py:106: UserWarning: Pandas requires version '2.6.2' or newer of 'numexpr' (version '2.6.1' currently installed).
  warnings.warn(msg, UserWarning)
            sugar
quarter          
1990Q1   0.330000
1990Q2   0.316667
1990Q3   0.246667
1990Q4   0.216667
1991Q1   0.193333
   Unnamed: 0       upc  dma_code  ...  upc_dma_code     distance     sugar
0           0  10000001       529  ...  10000001 529   372.617769  0.376667
1           1  10000001       561  ...  10000001 561  1265.954500  0.290000
2           2  10000001       618  ...  10000001 618  1626.333591  0.376667
3           3  10000001       640  ...  10000001 640   850.837264  0.376667
4           4  10000001       652  ...  10000001 652   954.904401  0.290000

[5 rows x 38 columns]
         cocoa_beans
quarter             
2000Q1      0.895000
2000Q2      0.926667
2000Q3      0.913333
2000Q4      0.883333
2001Q1      1.003333
   Unnamed: 0       upc  dma_code  ...     distance     sugar  cocoa_beans
0           0  10000001       529  ...   372.617769  0.376667     1.556667
1           1  10000001       561  ...  1265.954500  0.290000     2.596667
2           2  10000001       618  ...  1626.333591  0.376667     1.556667
3           3  10000001       640  ...   850.837264  0.376667     1.556667
4           4  10000001       652  ...   954.904401  0.290000     2.596667

[5 rows x 39 columns]
Traceback (most recent call last):
  File "/home/zcw7438/.local/lib/python3.6/site-packages/pandas/core/ops/__init__.py", line 1254, in na_op
    result = op(x, y)
  File "/home/zcw7438/.local/lib/python3.6/site-packages/pandas/core/ops/roperator.py", line 57, in ror_
    return operator.or_(right, left)
TypeError: ufunc 'bitwise_or' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zcw7438/.local/lib/python3.6/site-packages/pandas/core/ops/__init__.py", line 1269, in na_op
    result = libops.scalar_binop(x, y, op)
  File "pandas/_libs/ops.pyx", line 170, in pandas._libs.ops.scalar_binop
ValueError: Does not understand character buffer dtype format string ('?')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "logitEstimation.py", line 87, in <module>
    GenerateDEData(products, quarterOrMonth, inputs, characteristics)
  File "logitEstimation.py", line 33, in GenerateDEData
    add_features = AddExtraFeatures(products)
  File "logitEstimation.py", line 18, in AddExtraFeatures
    features_year['chocolate'] = np.where(features_year['variety_descr']=='CHOCOLATE' | (features_year['variety_descr'].isna()), 'CHOCOLATE', 'NON-CHOCOLATE')
  File "/home/zcw7438/.local/lib/python3.6/site-packages/pandas/core/ops/__init__.py", line 1321, in wrapper
    res_values = na_op(self.values, ovalues)
  File "/home/zcw7438/.local/lib/python3.6/site-packages/pandas/core/ops/__init__.py", line 1280, in na_op
    dtype=x.dtype, typ=type(y).__name__
TypeError: cannot compare a dtyped [bool] array with a scalar of type [bool]
